### ISSUE TITLE
Explore using Backpack for batched cg and Hessian inversion

### ISSUE THREAD

**Issue by Xuzzo** [appliedAI-Initiative/pyDVL#202]
Inverting the Hessian is a very computationally heavy process. At the moment, we dont parallelise it and purely rely on either the direct method (memory heavy) or scipy conjugate gradient implementation. In the past we tried implementing our own version of batched conjugate gradient, with several issues (see #137). 
[Backpack](https://docs.backpack.pt/en/master/index.html) is a pytorch library that extracts info from backpropagation. Among many things, they have a form of [batched gradient](https://docs.backpack.pt/en/master/extensions.html#backpack.extensions.BatchGrad) which could help us parallelise Hessian inversion. They also have a few optimised ways to [calculate the hessian directly](https://docs.backpack.pt/en/master/basic_usage/example_all_in_one.html#second-order-extensions). 
We could look into whether any of these speed up our workflow.

**Comment by mdbenito** [comment]
@Xuzzo What's the status here?
